plugins/UsersManager/NewsletterSignup.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link http://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+namespace Piwik\Plugins\UsersManager;
+
+use Exception;
+use Piwik\Config;
+use Piwik\Container\StaticContainer;
+use Piwik\Http;
+use Piwik\Option;
+use Piwik\Url;
+
+class NewsletterSignup
+{
+    public static function signupForNewsletter($userLogin, $email, $matomoOrg = false, $professionalServices = false)
+    {
+        // Don't bother if they aren't signing up for at least one newsletter
+        if (! ($matomoOrg || $professionalServices)) {
```

Comment:
We will also need to return if `Piwik\SettingsPiwik::isInternetEnabled()` is disabled